projects/igniteui-angular/src/public_api.ts

```diff
@@ -19,7 +19,9 @@ export * from './lib/directives/drag-drop/public_api';
 export * from './lib/directives/filter/filter.directive';
 export * from './lib/directives/focus/focus.directive';
 export * from './lib/directives/focus-trap/focus-trap.directive';
-export * from './lib/directives/for-of/for_of.directive';
+export {
+    IgxForOfContext, IgxForOfDirective, IForOfState, IgxGridForOfContext, IgxGridForOfDirective
+} from './lib/directives/for-of/for_of.directive';
```

Comment:
The star export also included a random `getTypeNameForDebugging` function and `IForOfDataChangingEventArgs` for an internal event, which I skipped along with the new token
// CC @kdinev